lib/Sema/TypeCheckDecl.cpp

```diff
@@ -3812,9 +3814,13 @@ void TypeChecker::validateDecl(ValueDecl *D) {
 
   case DeclKind::TypeAlias: {
     auto typeAlias = cast<TypeAliasDecl>(D);
-    // Check generic parameters, if needed.
+
     DeclValidationRAII IBV(typeAlias);
-    validateTypealiasType(*this, typeAlias);
+
+    // Finally, set the interface type.
+    if (!typeAlias->hasInterfaceType())
```

Comment:
Is it ever the case that hasInterfaceType() returns true here? If you copy and pasted this from the old code it's just an artifact of validateDecl() being called after validateDeclForNameLookup()

Reply:
It was true when the code was written - something something cycles with witness matching.